### PR TITLE
Instances Aggregate and Search

### PR DESCRIPTION
**Pull request by doctrino**
### Added
- Support for searching and aggregating across instanceces in the Data Modeling API with the implementation 
  `client.data_modeling.instances`, the methods `search` and `aggregate`.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Comment by erlendvollset** [comment]
I rebased this on #1265 and fixed the typing issues. Then I made some changes in the following commits. LGTM, but hold off with merging until 1265 goes in since I change the base branch to that one.

File: cognite/client/_api/data_modeling/instances.py [hunk 1]
```diff
@@ -576,25 +576,23 @@ def search(
     def aggregate(
         self,
         view: ViewId,
+        aggregates: Aggregation | Sequence[Aggregation],
```
[the remark this answers is not in the document]
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/instances.py:579]: Omitting aggregates only makes sense in combination with groupBy, and is used for retrieving all unique values. It's a clunky API which is quite overloaded, so just narrowing down what parts of the search API we use in this specific method. If we want to enable the user to get all unique values, we can make a dedicated method for that, but it's a really expensive operation when done in the way this endpoint does it, so i'd prefer to not expose it just yet unless someone has a particular need for it.

File: cognite/client/_api/data_modeling/instances.py [hunk 2]
```diff
@@ -574,45 +573,17 @@ def search(
...
-    ) -> EdgeAggregationResultList:
-        ...
-
     def aggregate(
         self,
         view: ViewId,
         instance_type: Literal["node", "edge"] = "node",
         query: str | None = None,
-        aggregates: Aggregation | dict | list[Aggregation | dict] | None = None,
+        aggregates: Aggregation | Sequence[Aggregation] | None = None,
```
[the remark this answers is not in the document]
Comment (erlendvollset) [reply cognite/client/_api/data_modeling/instances.py]: Although I'm not really a huge fan of accepting the untyped objects.. Users will probably be surprised by that failing when the dict has snake cased keys, which I really don't want to have to support everywhere.

[the remark this answers is not in the document]
Comment (doctrino) [reply cognite/client/_api/data_modeling/instances.py]: agree with that, but I think it is more for consistency with other endpoints.

**Review by erlendvollset — APPROVED** [review]
LGTM!

**Pull request merged**